conans/cli/api/subapi/remotes.py

```diff
@@ -64,20 +62,48 @@ def rename(self, remote: Remote, new_name: str):
     @api_method
     def user_info(self, remote: Remote):
         app = ConanApp(self.conan_api.cache_folder)
+        # FIXME: The return of this should be a model, not a dict
+        # FIXME: This is weird, probably the storage of users should all be in the json
+        # FIXME: If the password/token is kept in the model, this method is USELESS, just read
+        #        the remote and check if credentials and instead of the custom dict, just remote
+        #        objects can be returned
+        if remote.generic:
+            return {"name": remote.name,
+                    "authenticated": (remote.password is not None and remote.username is not None),
+                    "user_name": remote.username
+                    }
         return users_list(app.cache.localdb, remotes=[remote])[0]
 
     @api_method
     def login(self, remote: Remote, username, password):
         app = ConanApp(self.conan_api.cache_folder)
-        app.remote_manager.authenticate(remote, username, password)
+        # FIXME: This is weird, probably the storage of users should all be in the json
+        if remote.generic:
+            remote.username = username
+            remote.password = password
+            app.cache.remotes_registry.update(remote)
```

Comment:
But first, of that, we need to choose:
  a)  Need to consider that the "--generic"  is only "artifactory".
  b)  Introduce the `--artifactory-generic` type.

Because now you can use this feature for any remote that accepts basic http.

Reply:
Sure, makes sense. Lets start with the fully generic, then we'll see.